src/adapter/src/coord.rs

```diff
@@ -2362,12 +2359,14 @@ pub fn serve(
         // use the same impl!
         let timestamp_oracle_impl = catalog.system_config().timestamp_oracle_impl();
 
-        let initial_timestamps = get_initial_oracle_timestamps(
-            &catalog,
-            &timestamp_oracle_url,
-            &timestamp_oracle_metrics,
-        )
-        .await?;
+        let timestamp_oracle_metrics =
+            Arc::new(timestamp_oracle::metrics::Metrics::new(&metrics_registry));
+        let pg_timestamp_oracle_config = timestamp_oracle_url.map(|pg_url| {
+            PostgresTimestampOracleConfig::new(&pg_url, Arc::clone(&timestamp_oracle_metrics))
```

Comment:
We could instead make `PostgresTimestampOracleConfig::new` take a MetricsRegistry and internally construct the Metrics. That would save a bit of public API surface area

Reply:
yep, that refactor just didn't go far enough